File: tests/fixtures/anti-patterns/csharp/GodObjectVulnerable.cs
/**
 * Test Fixture: God Object Vulnerable Pattern (C#)
 *
 * This class has too many public methods (10+), indicating a god object.
 * Expected detections: >=1 violation for AP-001
 * Rule ID: AP-001
 * Severity: HIGH (warning)
 */

using System;

namespace TestFixtures.AntiPatterns
{
    public class GodObjectVulnerable
    {
        private string _data;

        public void Method01_ProcessInput() { }
        public void Method02_ValidateData() { }
        public void Method03_TransformData() { }
        public void Method04_SaveToDatabase() { }
        public void Method05_LoadFromDatabase() { }
        public void Method06_SendNotification() { }
        public void Method07_GenerateReport() { }
        public void Method08_ExportToCsv() { }
        public void Method09_ExportToJson() { }
        public void Method10_CalculateMetrics() { }
        public void Method11_AggregateResults() { }
        public void Method12_FilterItems() { }
    }
}

File: tests/fixtures/anti-patterns/typescript/god-object-vulnerable.ts
/**
 * Test Fixture: God Object Vulnerable Pattern (TypeScript)
 *
 * This class has too many methods (10+), indicating a god object.
 * Expected detections: >=1 violation for AP-001
 * Rule ID: AP-001
 * Severity: HIGH (warning)
 */

export class GodObjectVulnerable {
  private data: any;

  method01ProcessInput() { return; }
  method02ValidateData() { return; }
  method03TransformData() { return; }
  method04SaveToDatabase() { return; }
  method05LoadFromDatabase() { return; }
  method06SendNotification() { return; }
  method07GenerateReport() { return; }
  method08ExportToCsv() { return; }
  method09ExportToJson() { return; }
  method10CalculateMetrics() { return; }
  method11AggregateResults() { return; }
  method12FilterItems() { return; }
}
